article/article.md

```diff
@@ -0,0 +1,333 @@
+            artifact = "io.grpc:protoc-gen-grpc-java:$grpcVersion"
+        }
+    }
+    generateProtoTasks {
+        ofSourceSet("main").forEach {
+            it.plugins { grpc {} }
+        }
+    }
+}
+```
+
+gRPC provides classes such as `Server` and `ServerBuilder` which allow us to start a simple service in just a few lines of code. The most important part is to implement the abstract method `getPokemon` inside the generated class `PokedexGrpc.PokedexImplBase` in our case, delivering the data requested by client.
+
+Here is an extract of the Server implementation:
+
+```Kotlin
+class PokedexServer {
+
+    private val port = 50052
+    private var server: Server? = null
+
+    @Throws(IOException::class)
+    private fun start() {
+        //...
+        
+        server = ServerBuilder.forPort(port)
+            .addService(PokedexImpl())
+            .build()
+            .start()
+    
+        Runtime.getRuntime().addShutdownHook(object : Thread() {
+            override fun run() {
+                System.err.println("*** shutting down gRPC server since JVM is shutting down")
+                this@PokedexServer.stop()
+                System.err.println("*** server shut down")
+            }
+        })
+    }
+
+    private fun stop() {
+        server?.shutdown()
+    }
+
+    internal class PokedexImpl() : PokedexGrpc.PokedexImplBase() {
+
+        override fun getPokemon(request: PokedexRequest, responseObserver: StreamObserver<PokedexReply>) {
+            
+            val englishName = request.englishName
+            
+            // logic of find the pokemon info...
+
+            val reply = PokedexReply.newBuilder()
+                    .setId("pokemon ID" )
+                    .setFrenchName("pokemon French name")
+                    .setType("pokemon type")
+                    .setImageUrl("pokemon image URL")
+                    .build()
+        
+            responseObserver.onNext(reply)
+            responseObserver.onCompleted()
+        }
+    }
+
+    companion object {
+        @Throws(IOException::class, InterruptedException::class)
+        @JvmStatic
+        fun main(args: Array<String>) {
+            val server = PokedexServer()
+            server.start()
+            server.blockUntilShutdown()
+        }
+    }
+
+}
+```
+
+Once you finish implementing the server side logic, you can write a test to see if the server returns the expected value. There are examples that you can find on [github](https://github.com/grpc/grpc-java/tree/master/examples). gRPC provides classes such as `GrpcCleanupRule`, `InProcessServerBuilder` to make unit testing easier.
+
+You can check [here](https://github.com/xebia-france/hello-grpc/blob/master/serverapp/src/main/kotlin/fr/xebia/hellogrpc/PokedexServer.kt) to see the full implementation of `PokedexServer` and [here](https://github.com/xebia-france/hello-grpc/blob/master/serverapp/src/test/kotlin/fr/xebia/hellogrpc/PokedexServerTest.kt) for its unit test.
```

Comment:
For SEO efficiency, you should not reference `here` but a meaning group of words